pkg/uninstall/uninstall.go

```diff
@@ -65,7 +70,16 @@ func (uninstaller *uninstaller) delete(ctx context.Context, pkg ctrlpkg.Package,
 	case *v1alpha1.ClusterPackage:
 		return uninstaller.client.ClusterPackages().Delete(ctx, pkg, deleteOptions)
 	case *v1alpha1.Package:
-		return uninstaller.client.Packages(pkg.Namespace).Delete(ctx, pkg, deleteOptions)
+		var validateDeletion bool
+		if deleteNamespace {
+			validateDeletion = ValidateNamespaceDeletion(ctx, pkg)
+		}
+		err := uninstaller.client.Packages(pkg.Namespace).Delete(ctx, pkg, deleteOptions)
+		if validateDeletion {
+			DeleteNamespace(ctx, pkg.GetNamespace())
+		}
+		return err
```

Comment:
namespace deletion should be done in `UninstallBlocking`, after awaitDeletion has finished without error. In addition to that, the operation should wait for the namespace to be deleted, similar to the package